lib/modules/datasource/docker/index.spec.ts

```diff
@@ -562,6 +562,399 @@ describe('modules/datasource/docker/index', () => {
+            manifests: [
+              {
+                digest: currentDigest,
+                platform: {},
+              },
+            ],
+          },
+          {
+            'content-type': 'text/plain',
+          }
+        );
+      httpMock
+        .scope(baseUrl)
+        .get('/')
+        .twice()
+        .reply(401, '', {
+          'www-authenticate':
+            'Bearer realm="https://auth.docker.io/token",service="registry.docker.io",scope="repository:library/some-dep:pull  "',
+        })
+        .head('/library/some-dep/manifests/some-new-value')
+        .reply(200, undefined, {})
+        .get('/library/some-dep/manifests/some-new-value')
+        .reply(
+          200,
+          {
+            schemaVersion: 2,
+            mediaType: MediaType.manifestListV2,
+            manifests: [
+              {
+                digest:
+                  'sha256:c3fe2aac7e4f47270eeff0fdd35cb9bad674105eaa1663942645ca58399a2dbc',
+                platform: {
+                  architecture: 'arm',
+                  os: 'linux',
+                  variant: 'v6',
+                },
+              },
+              {
+                digest:
+                  'sha256:78fa4d63fec4e647f00908f24cda05af101aa9702700f613c7f82a96a267d801',
+                platform: {
+                  architecture: '386',
+                  os: 'linux',
+                },
+              },
+              {
+                digest:
+                  'sha256:81093b981e72a54d488d5a60780006d82f7cc02d248d88ff71ff4137b0f51176',
+                platform: {
+                  architecture: 'amd64',
+                  os: 'linux',
+                },
+              },
+            ],
+          },
+          {
+            'content-type': 'text/plain',
+          }
+        );
+
+      const res = await getDigest(
+        {
+          datasource: 'docker',
+          depName: 'some-dep',
+          currentValue,
+          currentDigest,
+        },
+        'some-new-value'
+      );
+
+      expect(logger.logger.debug).toHaveBeenCalledWith(
+        `Current digest ${currentDigest} relates to architecture null`
+      );
+      expect(res).toBe(
+        'sha256:ee75deb1a41bb998e52a116707a6e22a91904cba0c1d6e6c76cf04923efff2d8'
+      );
+    });
+
+    it('supports architecture-specific digest in OCI manifests with media type', async () => {
+      const currentValue = 'some-current-value';
+      const currentDigest = 'some-image-digest';
+
+      httpMock
+        .scope(authUrl)
+        .get(
+          '/token?service=registry.docker.io&scope=repository:library/some-dep:pull'
+        )
+        .times(3)
+        .reply(200, { token: 'some-token' });
+      httpMock
+        .scope(baseUrl)
+        .get('/')
+        .twice()
+        .reply(401, '', {
+          'www-authenticate':
+            'Bearer realm="https://auth.docker.io/token",service="registry.docker.io",scope="repository:library/some-dep:pull  "',
```

Comment:
```suggestion
            'Bearer realm="https://auth.docker.io/token",service="registry.docker.io",scope="repository:library/some-dep:pull"',
```
there shouldn't be trailing spaces